instructor/common.py

```diff
@@ -0,0 +1,604 @@
+
+
+def course_delete(request, pk):
+    print "DELETE {}".format(pk)
+    models.Course.objects.get(id=pk).delete()
+    return redirect('common_course')
+
+
+def get_account_type(request):
+    account_type = ''
+    if request.user.id and len(request.user.groups.all()) > 0:
+        account_type = request.user.groups.all()[0].name
+    return account_type
+
+
+def assignments(request):
+
+    account_type = get_account_type(request)
+    public_list = models.Assignment.objects.filter(access='Public')
+
+    return render_to_response('instructor/assignments.html',
+                              {'assignments': public_list},
+                              context_instance=RequestContext(request))
+
+
+def assignments_delete(request, pk):
+    models.Assignment.objects.get(id=pk).delete()
+    return redirect('common_assignments')
+
+def create_new_assignment(request):
+    request.session['new'] = True
+    request.session['assignment_name'] = ''
+    request.session['based_on'] = ''
+    return redirect('common_assignment_setup')
+
+
+def assignment_setup(request):
+    if request.method == "POST":
+        #Have to give error for empty name
+        request.session['assignment_name'] = request.POST['name']
+        if request.POST['based_on']:
+            request.session['based_on'] = request.POST['based_on']
+
+        if 'continue' in request.POST:
+            return redirect('common_course_setup')
+
+    all_assignments = models.Assignment.objects.all()
+    return render_to_response('instructor/assignment_setup.html',
+                              {'assignments': all_assignments,
+                               'assignment_name': request.session['assignment_name'],
+                               'based_on': request.session['based_on'],
+                               'new': request.session['new']},
+                                context_instance=RequestContext(request))
+
+
+def course_setup(request):
+    CourseForm = modelform_factory(models.Course, fields=['code', 'name'])
+
+    if request.method == "POST":
+        if 'back' in request.POST:
+            return redirect("common_assignment_setup")
+        elif 'course_pk' in request.POST:
+            '''Adding to an existing course'''
+            course = models.Course.objects.get(pk=request.POST['course_pk'])
+        else:
+            '''Create a new course'''
+            user = User.objects.get(username=request.user)
+            form = CourseForm(request.POST)
+            course = form.save(commit=False)
+            course.owner_id = user.id
+            course.save()
+
+        '''Create a new assignment'''
+        assignment_name = request.session['assignment_name']
+        assignment_id = assignment_name + datetime.datetime.now().strftime("%I:%M%p on %B %d, %Y")
+
+        if request.session['based_on']:
+            a = models.Assignment(course=course,
+                              name=assignment_name,
```

Comment:
This indentation is a bit funny, please fix, I think probably
```python
a = models.Assignment(
    course=course,
    name=assignment_name,
    assignment_id=assignment_id,
    basedOn=request.session['based_on']
)
```
is probably the clearest.  Same comment for the else below